hips/hip-9999.md

```diff
@@ -0,0 +1,85 @@
+---
+hip: "9999"
+title: "Output Helm Chart Schemas"
+authors: [ "Remco Haszing <remcohaszing@gmail.com>" ]
+created: "2021-07-14"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+Helm supports JSON schema validation to validate `values.yaml`.
+It would be nice if JSON schema validation is also possible by editors of `values.yaml` files.
+It would be nice if JSON schemas can be referenced by dependant Helm chart JSON schemas.
+Both will be possible if the JSON schemas are served on separate URLs.
+
+## Motivation
+
+For Helm users it would be nice if they can reference JSON schemas from a Helm chart.
+
+If a Helm chart depends on another Helm chart, its JSON schema can reference the original JSON schema like so:
+
+```json
+{
+  "properties": {
+    "my-chart": {
+      "$ref": "https://charts.example/my-chart-1.2.3.schema.json"
+    }
+  }
+}
+```
+
+If a user wants to enable JSON schema validation and completion for `values.yaml` when using [`yaml-language-server`](https://github.com/redhat-developer/yaml-language-server), this can be accomplished by a modeline:
+
+```yaml
+# yaml-language-server: $schema=https://charts.example/my-chart-1.2.3.schema.json
+```
+
+## Rationale
+
+By outputting the JSON schema next to the packaged Helm chart, Helm determines the standard output location.
+Chart publishers could copy their JSON schema manually, but this isn’t standardized in any way.
+If Helm determines the standard output location of such files, chart publishers are likely to use this.
+Helm already outputs an additional file for signed charts. This proposal is consistent with that behaviour.
```

Comment:
This implies that chart publishers will publish schema files, but it does not explain how charts stored in OCI registries would make schemata available. While OCI is considered an experimental feature, because it is part of Helm's core we would need to introduce a solution for that registry as well as for a regular Helm chart repo.